### PR TITLE
End to End Testing

### LINKED ISSUE #37: End-to-End Testing
As part of our CI/CD, we need to include some form of end-to-end testing. This will be done using Cypress and integrated into our git actions.

A limitation that might pop up is needing to access the firebase database to log-in, which may cause issues due to the limited amount of accesses our database can have. Will also look into ways to get around this limitation.

### PR DESCRIPTION
**Pull request by DanielDowling999**
Adds end-to-end testing for the majority of the app, testing it in a way that users would. Also adds adjusts RTL tests to focus on components, deleting some unnecessary ones and adding a new test for the DateTime. Finally, it integrates the End-to-End Testing into the pipeline, automatically running the tests.

Taskboard is not covered by these tests yet, but an issue has been made for it.

Closes #37

### CODE REVIEW

**Review by thakurtushar02 — APPROVED** [review]

File: src/components/Navbar.jsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ function Navbar() {
           </div>
         </Link>
         <Link to="/board" className="button-link">
-          <div className="navbar__icon__group" data-testid="nav-trello-icon">
+          <div className="navbar__icon__group" data-testid="nav-taskboard-icon">
```
Comment (saaaaraaaah) [inline src/components/Navbar.jsx:44]: Nice spotting!

**Review by lucas2005gao — APPROVED** [review]
Thanks for all the hard work Daniel!

**Review by saaaaraaaah — APPROVED** [review]
Awesome work! LGTM!

**Comment by lucas2005gao** [comment]
Lets see if the e2e test Github Actions workflow runs fine

**Pull request merged**